src/query/models/types.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package models
+
+import (
+	"regexp"
+)
+
+// Separators for tags.
+const (
+	sep = byte(',')
+	eq  = byte('=')
+)
+
+// TagOptions describes additional options for tags.
+type TagOptions interface {
+	// Validate validates these tag options.
+	Validate() error
+
+	// SetMetricName sets the name for the `metric name` metric.
+	SetMetricName(metricName []byte) TagOptions
+
+	// GetDefaultName gets the name for the `metric name `metric`.
+	GetMetricName() []byte
```

Comment:
nit: Usually we don't prefix the getter methods with `Get` if you notice re: other places we have Options interfaces.

Reply:
Huh, if anything I'm more surprised that the linter passed considering the comment is `GetDefaultName...`

Will update